src/collection.rs

```diff
@@ -387,6 +420,33 @@ impl<T: Id<T>> CollectionWithId<T> {
             }
         }
     }
+
+    /// Merge a `CollectionWithId` parameter into the current one. Fails if any identifier into the
+    /// `CollectionWithId` parameter is already in the collection.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use navitia_model::collection::*;
+    /// # fn run() -> navitia_model::Result<()> {
+    /// # #[derive(PartialEq, Debug)] struct Obj(&'static str);
+    /// # impl Id<Obj> for Obj { fn id(&self) -> &str { self.0 } }
+    /// let mut c1 = CollectionWithId::new(vec![Obj("foo"), Obj("bar")])?;
+    /// let mut c2 = CollectionWithId::new(vec![Obj("foo"), Obj("qux")])?;
+    /// let mut c3 = CollectionWithId::new(vec![Obj("corge"), Obj("grault")])?;
+    /// assert!(c1.merge(c2).is_err());
+    /// c1.merge(c3);
+    /// assert_eq!(c1.len(), 4);
+    /// # Ok(())
+    /// # }
+    /// # fn main() { run().unwrap() }
+    /// ```
+    pub fn merge(&mut self, other: Self) -> Result<()> {
+        for item in other {
+            self.push(item)?;
+        }
+        Ok(())
```

Comment:
```rust
pub fn merge(&mut self, other: Self) -> Result<()> {
    other.into_iter().map(|item| self.push(item)).collect()
}
```

Reply:
hum i have this error when i try it:
```
   --> src/collection.rs:445:55
    |
445 |         other.into_iter().map(|item| self.push(item)).collect()
    |                                                       ^^^^^^^ a collection of type `()` cannot be built from an iterator over elements of type `collection::Idx<T>`
```

Reply:
self.push(item) returns an Idx not ()

Reply:
proposed code modified

Reply:
`for` is easier to understand

Reply:
As you wish, that's a non blocking comment.